### DISCUSSION TITLE
Get cost of a mode solver simulation

### DISCUSSION THREAD

**Discussion by samuelkim16** [flexcompute/tidy3d#1883] [Q&A] [answered]
Is there a way to get the cost of a mode solver simulation? 

Running `web.real_cost('task_id')` gives the message

```
08:49:00 Eastern Daylight Time WARNING: Billed FlexCredit for task              
                               'task_id' is   
                               not available. If the task has been successfully 
                               run, it should be available shortly.```

**Comment by momchil-flex** [comment, marked as answer]
@samuelkim16 our mode solver is still using some legacy handling including a web api that is separate from our other solvers, and some things unfortunately don't work (well). It is currently being rewritten to unify with the rest of the code (fyi @caseyflex ).

In the meantime, I found a way to get the real cost of a mode solver task after it has been completed, if you know the `task_id` and `solver_id` (printed when the task is submitted):

```
from tidy3d.web.api.mode import ModeSolverTask
task = ModeSolverTask.get(task_id=task_id, solver_id=solver_id)
print(task.real_flex_unit)
```

Or to do this even more directly without having to re-download the data file,

```
from tidy3d.web.core.http_util import http
resp = http.get(f"tidy3d/modesolver/py/{task_id}/{solver_id}")
print(resp["charge"])
```

Of course, you can also just go to our web GUI and see the task cost there, too.

**Reply by samuelkim16** [reply]
Thank you! Out of curiosity, is there a way to get `task_id` and `solver_id` programmatically? I couldn’t find an analogous method to `job.task_id`.

**Reply by caseyflex** [reply]
Once it is unified with the rest of the code, you can use the same convenient web api as for the other solvers. For now, I suppose you could use the following workaround, using `ModeSolverTask` directly to manage the task. I think the `task_id` and `solver_id` are set immediately following `ModeSolverTask.create`.

```
from tidy3d.web.api.mode import ModeSolverTask
import time
task = ModeSolverTask.create(mode_solver)
task.upload()
task.submit()
prev_status = "draft"
status = task.status
while status not in ("success", "error", "diverged", "deleted"):
    if status != prev_status:
        print(f"Mode solver status: {status}")
        prev_status = status
    time.sleep(0.5)
    status = task.get_info().status
print(task.get_info().real_flex_unit)
mode_solver_data = task.get_result()


task_id = task.task_id
solver_id = task.solver_id
